rotkehlchen/data/importer.py

```diff
@@ -469,3 +480,269 @@ def import_cryptocom_csv(self, filepath: Path) -> Tuple[bool, str]:
+                except DeserializationError as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'Error during BlockFi CSV import deserialization. '
+                        f'Error was {str(e)}. Ignoring entry',
+                    )
+                    continue
+                except UnsupportedBlockFiEntry as e:
+                    self.db.msg_aggregator.add_warning(str(e))
+                    continue
+                except KeyError as e:
+                    return False, str(e)
+        return True, ''
+
+    def _consume_blockfi_trade(self, csv_row: Dict[str, Any]) -> None:
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr='%Y-%m-%d %H:%M:%S',
+            location='BlockFi',
+        )
+
+        buy_asset = symbol_to_asset_or_token(csv_row['Buy Currency'])
+        buy_amount = deserialize_asset_amount(csv_row['Buy Quantity'])
+        sold_asset = symbol_to_asset_or_token(csv_row['Sold Currency'])
+        sold_amount = deserialize_asset_amount(csv_row['Sold Quantity'])
+        rate = Price(buy_amount / sold_amount)
+        trade = Trade(
+            timestamp=timestamp,
+            location=Location.BLOCKFI,
+            base_asset=buy_asset,
+            quote_asset=sold_asset,
+            trade_type=TradeType.BUY,
+            amount=buy_amount,
+            rate=rate,
+            fee=None,  # BlockFI doesn't provide this information
+            fee_currency=None,
+            link='',
+            notes=csv_row['Type'],
+        )
+        self.db.add_trades([trade])
+
+    def import_blockfi_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
+        """
+        Information for the values that the columns can have has been obtained from
+        the issue in github #1674
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for row in data:
+                try:
+                    self._consume_blockfi_trade(row)
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During BlockFi CSV import found action with unknown '
+                        f'asset {e.asset_name}. Ignoring entry',
+                    )
+                    continue
+                except DeserializationError as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'Error during BlockFi CSV import deserialization. '
+                        f'Error was {str(e)}. Ignoring entry',
+                    )
+                    continue
+                except UnsupportedBlockFiEntry as e:
+                    self.db.msg_aggregator.add_warning(str(e))
+                    continue
+                except KeyError as e:
+                    return False, str(e)
+        return True, ''
+
+    def _consume_nexo(self, csv_row: Dict[str, Any]) -> None:
```

Comment:
Please add a docstring with what this does AND importantly which exceptions it may raise